### PR TITLE
Add conflicts to rust and cargo-completion

### PR DESCRIPTION
**Pull request by J-M0**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/cargo-completion.rb [hunk 1]
```diff
@@ -8,6 +8,8 @@ class CargoCompletion < Formula
 
   bottle :unneeded
 
+  conflicts_with "rust"
```
Comment (neutric) [inline Formula/cargo-completion.rb]: Please add a reason, see https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md#specifying-conflicts-with-other-formulae.

File: Formula/rust.rb [hunk 2]
```diff
@@ -41,6 +41,8 @@ class Rust < Formula
   depends_on "openssl"
   depends_on "libssh2"
 
+  conflicts_with "cargo-completion"
```
Comment (neutric) [inline Formula/rust.rb]: See above.

**Comment by ilovezfs** [comment]
Is this intended for people using non-Homebrew rust?

**Comment by j-m0** [comment]
I would guess so. I don't know why cargo-completion is separate from rust, but they can't be installed at the same time.

**Comment by fxcoudert** [comment]
Thanks @J-M0 for the pull request!

**Pull request closed without merging**